python/tests/arithmetic_tests/test_tables.py

```diff
@@ -0,0 +1,19 @@
+from tests.aggregate_tests.aggtst_base import TstTable
+
+
+class arithtst_date_table(TstTable):
+    """Define the table used by DATE tests"""
+
+    def __init__(self):
+        self.sql = """CREATE FUNCTION d()
+                      RETURNS DATE NOT NULL AS
+                      CAST('1970-01-01' AS DATE);
+
+                      CREATE TABLE date_tbl(
+                      id INT,
+                      c1 DATE NOT NULL,
+                      c2 DATE)"""
+        self.data = [
```

Comment:
for dates/timestamps we will need to have wider coverage.
we need dates before 1970, and also dates before the Gregorian Calendar, i.e., Oct 1582.
moreover, dates before year 1 are illegal, and we do not support dates after 9999, so we will have to see what we do about negative tests as well.